services/streammanager/wunderkind/wunderkindmanager_test.go

```diff
@@ -0,0 +1,136 @@
+
+	// Empty payload
+	sampleEventJson, _ = json.Marshal(map[string]interface{}{
+		"payload": "",
+	})
+	statusCode, statusMsg, respMsg = producer.Produce(sampleEventJson, map[string]string{})
+	require.Equal(t, http.StatusBadRequest, statusCode)
+	require.Equal(t, "Failure", statusMsg)
+	require.Contains(t, respMsg, "[Wunderkind] error :: Invalid payload")
+}
+
+func TestProduceWithServiceResponse(t *testing.T) {
+	conf := config.New()
+	conf.Set("WUNDERKIND_REGION", "us-east-1")
+	conf.Set("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
+	conf.Set("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
+	conf.Set("WUNDERKIND_LAMBDA", sampleFunction)
+
+	ctrl := gomock.NewController(t)
+	mockClient := mock_lambda.NewMockLambdaClient(ctrl)
+	mockLogger := mock_logger.NewMockLogger(ctrl)
+	producer := &Producer{conf: conf, client: mockClient, logger: mockLogger}
+
+	sampleEventJson, _ := json.Marshal(map[string]interface{}{
+		"payload": sampleMessage,
+	})
+
+	destConfig := map[string]string{}
+
+	var sampleInput lambda.InvokeInput
+	sampleInput.SetFunctionName(sampleFunction)
+	sampleInput.SetPayload([]byte(sampleMessage))
+	sampleInput.SetInvocationType(invocationType)
+	sampleInput.SetLogType("Tail")
+
+	mockClient.EXPECT().Invoke(&sampleInput).Return(&lambda.InvokeOutput{}, nil)
+	statusCode, statusMsg, respMsg := producer.Produce(sampleEventJson, destConfig)
+	require.Equal(t, http.StatusOK, statusCode)
+	require.Equal(t, "Success", statusMsg)
+	require.NotEmpty(t, respMsg)
+
+	// return general Error
+	errorCode := "errorCode"
+	mockClient.EXPECT().Invoke(&sampleInput).Return(nil, errors.New(errorCode))
+	mockLogger.EXPECT().Warnn(gomock.Any(), gomock.Any()).Times(1)
+	statusCode, statusMsg, respMsg = producer.Produce(sampleEventJson, destConfig)
+	require.Equal(t, http.StatusInternalServerError, statusCode)
+	require.Equal(t, "Failure", statusMsg)
+	require.NotEmpty(t, respMsg)
+
+	// return an error when lambda invocation is successful, but there is an issue with the payload
+	mockClient.EXPECT().Invoke(&sampleInput).Return(&lambda.InvokeOutput{
+		StatusCode:      aws.Int64(http.StatusOK),
+		FunctionError:   aws.String("Unhandled"),
+		ExecutedVersion: aws.String("$LATEST"),
+	}, nil)
+	mockLogger.EXPECT().Warnn(gomock.Any(), gomock.Any()).Times(1)
+	statusCode, statusMsg, _ = producer.Produce(sampleEventJson, destConfig)
+	require.Equal(t, http.StatusBadRequest, statusCode)
+	require.Equal(t, "Failure", statusMsg)
+
+	// return aws error
+	mockClient.EXPECT().Invoke(&sampleInput).Return(
+		nil, awserr.NewRequestFailure(awserr.New(errorCode, errorCode, errors.New(errorCode)), http.StatusBadRequest, "request-id"))
+	mockLogger.EXPECT().Warnn(gomock.Any(), gomock.Any()).Times(1)
+	statusCode, statusMsg, respMsg = producer.Produce(sampleEventJson, destConfig)
+	require.Equal(t, http.StatusBadRequest, statusCode)
+	require.Equal(t, errorCode, statusMsg)
+	require.NotEmpty(t, respMsg)
```

Comment:
For different tests cases, we can use independent `t.Run()` which provides better isolation for the test case. Also gives feasibiilty to run individual test case. Something like this:
```GO
	t.Run("general error", func(t *testing.T) {
		errorCode := "errorCode"
		mockClient.EXPECT().Invoke(&sampleInput).Return(nil, errors.New(errorCode))
		mockLogger.EXPECT().Warnn(gomock.Any(), gomock.Any()).Times(1)
		statusCode, statusMsg, respMsg = producer.Produce(sampleEventJson, destConfig)
		require.Equal(t, http.StatusInternalServerError, statusCode)
		require.Equal(t, "Failure", statusMsg)
		require.NotEmpty(t, respMsg)
	})
	t.Run("when lambda invocation is successful, but there is an issue with the payload", func(t *testing.T) {
		mockClient.EXPECT().Invoke(&sampleInput).Return(&lambda.InvokeOutput{
			StatusCode:      aws.Int64(http.StatusOK),
			FunctionError:   aws.String("Unhandled"),
			ExecutedVersion: aws.String("$LATEST"),
		}, nil)
		mockLogger.EXPECT().Warnn(gomock.Any(), gomock.Any()).Times(1)
		statusCode, statusMsg, _ = producer.Produce(sampleEventJson, destConfig)
		require.Equal(t, http.StatusBadRequest, statusCode)
		require.Equal(t, "Failure", statusMsg)
	})
	t.Run("aws error", func(t *testing.T) {
		errorCode := "errorCode"
		mockClient.EXPECT().Invoke(&sampleInput).Return(
			nil, awserr.NewRequestFailure(awserr.New(errorCode, errorCode, errors.New(errorCode)), http.StatusBadRequest, "request-id"))
		mockLogger.EXPECT().Warnn(gomock.Any(), gomock.Any()).Times(1)
		statusCode, statusMsg, respMsg := producer.Produce(sampleEventJson, destConfig)
		require.Equal(t, http.StatusBadRequest, statusCode)
		require.Equal(t, errorCode, statusMsg)
		require.NotEmpty(t, respMsg)
	})
```